docs/source/usage/contributing.rst

```diff
@@ -1,5 +1,175 @@
-Feel free to come and open new PRs or issues.
\ No newline at end of file
+PyDESeq2 is a living project and any contributions are welcome!
+The project is hosted on `Github <https://github.com/owkin/PyDESeq2>`_.
+
+You can help us by adding a star to `pydeseq` repository on github or talk about
+our repository on the social media or on your blog or article.
+
+We also kindly ask you to follow `Python code of conduct <https://www.python.org/psf/codeofconduct/>`_.
+
+If you find any bugs in the code report them to the
+`issue tracker <https://github.com/owkin/PyDESeq2/issues>`_.
+
+How can I contribute?
+=====================
+- :ref:`bug`
+- :ref:`code_contrib`
+- :ref:`doc_contrib`
+- :ref:`answers`
+
+.. _bug:
+
+Submit a bug or ask for a new feature
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+First check if the issue which you are facing or a feature you would like to
+have added is not already described in the
+`issue tracker <https://github.com/owkin/PyDESeq2/issues>`_.
+If this is the case you can give a thumb up to that
+issue / feature. If there are multiple thumbs ups to a given issue it makes us
+prioritize it higher. You can also add a comment if you think there is something
+you would like to add to what have been said.
+
+If the issue / feature is not yet described make sure to give precise details on
+how you reached that issue (if possible giving code snippets) or precisely
+describe how the new feature should function. If it's an issue you should also
+mention which version of `pydeseq2` you are using. If you don't know which
+version you are using you can check it by running the following python code:
+
+.. code-block:: python
+
+    import pydeseq2
+    pydeseq2.__version__
+
+.. _code_contrib:
+
+Contribute with the code
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Thanks for willing to help us out with the `pydeseq2` development.
+
+If you are contributing to the existing issue add in the comment of that issue
+that you plan to contribute on it to avoid duplicate work. Once you start
+working on your pull request, link it to the issue by adding `#issue_number`
+
+1. To contribute you will first need to fork the `Github <https://github.com/owkin/PyDESeq2>`_
+repository using the fork button in github. This will create a copy of
+`pydeseq2` on your github account (you must be logged in to github).
+
+1. clone the `pydeseq2` version from your account:
+
+.. code-block:: python
+
+    git clone git@github.com:<your-github-username>/pydeseq2.git
+    cd pydeseq2
+
+We recommend using conda environment, run:
+
+.. code-block:: bash
+
+    conda env create -n pydeseq2-dev python=3.8
+    conda activate pydeseq2-dev
+
+and then install `pydeseq2` in the development mode. This will also install all
+the required dependencies.
+
+.. code-block:: bash
+
+    pip install -e ."[dev]"
+
+The pre-commit tool will automatically run `black` and `isort`, and check `flake8`
```

Comment:
Don't you also need to run `pre-commit install` before? (Maybe there's a way to automate this?)